### PR TITLE
14168 login signup after

### LINKED ISSUE #14168: Oauth flow: login/signup after-hook
Ensure that if you are not logged-in when redirected to the consent screen, you eventually get redirected to it.

- If the user opts to log in, take him to the login screen afterwards.
- If the user chooses signup, redirect him to auth after signup.

Let's focus only in organization login for now (no central involved yet).

### PR DESCRIPTION
**Pull request by javitonino**
Closes #14168 #14212 

We have removed from the scope that the signup redirects you back to the oauth flow after signup. It was too confusing for the user, and it's not easy to implement. We can revisit in the future.

So this:
- Makes the oauth URLs independent of subdomain or /u/ part. i.e: if you are logged in as user2, user1.carto.com or org.carto.com will still work if you are in the same cloud. Very useful for single-cloud installations. This basically meant using `current_viewer` instead of `current_user`
- Redirect tback to oauth from login already worked, I only added a test.
- Some signup fixes for free (styles, urls...)

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -212,6 +241,10 @@ def login_required
     is_auth ? validate_session(current_user) : not_authorized
   end
 
+  def any_login_required
```
Comment (ethervoid) [inline app/controllers/application_controller.rb]: Maybe change the name for `active_login_required`

File: app/views/carto/oauth_provider/consent.html.erb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Authorize?
 
-<%= form_tag(CartoDB.url(self, :oauth2_authorize, {}, current_user), method: :post) do %>
+<%= form_tag(CartoDB.url(self, :oauth2_authorize, {}, current_viewer), method: :post) do %>
```
Comment (ethervoid) [inline app/views/carto/oauth_provider/consent.html.erb]: Does this current_viewer come from the `application_helper`?

File: app/helpers/signup_helper.rb [hunk 3]
```diff
@@ -17,4 +17,10 @@ def google_sign_in?
   def github_sign_in?
     @user.github_user_id && @organization.auth_github_enabled
   end
+
+  def signup_url
```
Comment (ethervoid) [inline app/helpers/signup_helper.rb]: why do we need this?

**Comment by javitonino** [comment]
second :eyes: ?

**Comment by alrocar** [comment]
Acceptance 🍏 for login redirect.

@javitonino Should I be able to test `signup` in staging?

**Pull request merged**